storage/operation/pebbleimpl/seeker.go

```diff
@@ -0,0 +1,77 @@
+package pebbleimpl
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/utils/merr"
+
+	"github.com/cockroachdb/pebble"
+)
+
+type pebbleSeeker struct {
+	reader pebble.Reader
+}
+
+var _ storage.Seeker = (*pebbleSeeker)(nil)
+
+func newPebbleSeeker(reader pebble.Reader) *pebbleSeeker {
+	return &pebbleSeeker{
+		reader: reader,
+	}
+}
+
+// SeekLE (seek less than or equal) returns given key if present.  Otherwise,
+// it returns the largest key that is less than the given key in lexicographical
+// order within the prefix range [startPrefix, endPrefix], both inclusive.
+// This function returns error if given key is outside range of startPrefix and endPrefix.
+func (i *pebbleSeeker) SeekLE(startPrefix, endPrefix []byte, key []byte) (_ []byte, _ bool, errToReturn error) {
```

Comment:
```suggestion
func (i *pebbleSeeker) SeekLE(startPrefix, key []byte) (_ []byte, _ bool, errToReturn error) {
```

What about we remove the `endPrefix`? Since the key itself is actually the upper bound, and the lowerBound is always the startPrefix itself. 

This could also eliminates the user error case.

Reply:
> What about we remove the endPrefix?

Great idea! :+1: Done.